crates/vault-registry/src/lib.rs

```diff
@@ -1335,6 +1335,29 @@ impl<T: Config> Module<T> {
         }
     }
 
+    /// Get all vaults with non-zero issued (thus redeemable) tokens, ordered in descending order of this amount
+    pub fn get_vaults_with_redeemable_tokens() -> Result<Vec<(T::AccountId, Issuing<T>)>, DispatchError> {
+        // find all vault accounts with sufficient collateral
+        let vaults_with_redeemable_tokens = <Vaults<T>>::iter()
+            .filter_map(|account_id, _vault| {
+                let vault = Into::<RichVault<T>>::into(v.1);
+                let redeemable_tokens = vault.redeemable_tokens().ok()?;
+                if !redeemable_tokens.is_zero() {
+                    Some((account_id, redeemable_tokens))
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<(_, _)>>();
+
+        if vaults_with_redeemable_tokens.is_empty() {
+            Err(Error::<T>::NoVaultWithRedeemableTokens.into())
```

Comment:
A fix that we should probably do is to return an empty array instead of an error when there are 0 vaults. Because this is not an error state per see, it's just that we have 0 vaults right now. IMO we should also correct that behavior for issuable tokens.